.eslintrc.cjs

```diff
@@ -0,0 +1,252 @@
+                        constructors: 'no-public',                                  // ...require an explicit "protected" or "private" on constructors
+                    },
+                }],
+                '@typescript-eslint/return-await': ['error', 'always'],     // Allow return await
+                '@typescript-eslint/naming-convention': ['error', {    // Define naming convention rules
+                    selector: 'variable',
+                    modifiers: ['unused'],                             // Specifically target unused variables
+                    format: ['camelCase', 'PascalCase', 'UPPER_CASE'], // Use existing naming convention formatting
+                    leadingUnderscore: 'require',                      // Require a leading underscore, as otherwise unused vars fail standard the naming convention
+                }],
+                '@typescript-eslint/no-misused-promises': ['error',     // Overwrite from plugin:@typescript-eslint/recommended-requiring-type-checking
+                    {
+                        checksVoidReturn: false, // Allows async functions in void-returning places
+                    },
+                ],
+                '@typescript-eslint/no-unused-vars': ['error', {    // Disallow unused variables
+                    args: 'all',                                    // All function parameters/arguments must be used
+                    varsIgnorePattern: '^_',                        // Variables declared with names that start with an underscore can be unused
+                    argsIgnorePattern: '^_',                        // Function arguments declared with names that start with an underscore can be unused
+                }],
+            },
+            settings: {
+                'import/parsers': {
+                    '@typescript-eslint/parser': ['.ts'],
+                },
+                'import/resolver': {
+                    typescript: {
+                        alwaysTryTypes: true,           // Always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
+                        project: [
+                            './tsconfig.json',
+                        ],
+                    },
+                },
+            },
+        },
+        {
+            // Rules for Jest files
+            files: [
+                '**/*.spec.js',
+                '**/*.spec.ts',
+                '**/*.test.js',
+                '**/*.test.ts',
+            ],
+            env: {
+                jest: true,
+            },
+            plugins: [
+                'jest',
+            ],
+            extends: [
+                'plugin:jest/recommended',
+            ],
+            parser: '@typescript-eslint/parser',
+            rules: {
+                '@typescript-eslint/dot-notation': 'off', // Allow for object['property'] notation for testing of protected/private classes
+            },
+        },
+    ],
```

Comment:
Can you add a TODO here about possibly adding support for other testing frameworks such as mocha or vitest?